### ISSUE TITLE
QA data storage models

### ISSUE THREAD

**Issue by rosemm** [arcus/education_modules#47]
# Module Quality Assurance Report for PR #46 
----
Date: 2022-03-04
Reviewer: Rose Hartman
Name of Module: Types of Data Storage Models
Current Liascript URL: https://liascript.github.io/course/?https://raw.githubusercontent.com/arcus/education_modules/data-storage-models/data_storage_models/data_storage_models.md#1 
Current Version of Module (use the latest commit value):  https://github.com/arcus/education_modules/pull/46/commits/3547c8a95a10741297ec9c70cbfc8d80db1a5a21 

# Checklist Reports:

## Structural Elements

* [ ] YAML top section filled in with name, email, language, narrator, title, and comment (blurb) filled out appropriately.
* [ ] YAML top section includes proper link to CSS (currently https://chop-dbhi-arcus-education-website-assets.s3.amazonaws.com/css/styles.css).
* [ ] YAML top section has a version of at least 1.0.0 (first public version).
* [ ] Title is the first line and is the only level-1 header in the document.
* [ ] Overview section immediately follows Title, surrounded in div with class overview, and has filled in sections including a short and focused intro blurb, an "Is this module right for me?" section that includes a longer description written in simple language, Estimated time to completion, Pre-requisites, and Learning objectives.
* [ ] Contents within Overview reflect accurately the sections and the links in the contents section work.
* [ ] Sections following Overview all have content (no pages with just header and no additional text / media material).
* [ ] Final section is Feedback.

## Content

* [ ] Good amount of content, both in terms of the complexity/usefulness of the material covered and the time estimate
* [ ] Clearly defined learning objectives using strong, descriptive verbs. (See [Bloom's taxonomy](https://cft.vanderbilt.edu/guides-sub-pages/blooms-taxonomy/) for ideas.)
* [ ] Every learning objective is covered in the module content.
* [ ] There are no tangents or mission creep in the module content, straying from the learning objectives.
* [ ] No betrayal of expectations: The module title, description, learning objectives, time estimate, and overview all accurately reflect the content of the module. A learner should be able to make an informed decision about whether or not to complete the module.
* [ ] Avoids unclear language: unexplained idioms or references, unexplained acronyms, unnecessary technical language.
* [ ] Unusual words, or words taking on a very specific meaning in context, are always defined for the user, either on the page (e.g. using footnotes) or with links to a definition/glossary. Provides pronunciation guides for especially unusual words of particular importance.
* [ ] Provides content in a variety of forms and styles: screencasts, text, webinars/lectures, practical exercises, etc. Whenever possible, multiple forms/styles should be incorporated in each module so learners have multiple avenues to the content.
* [ ] Avoids unnecessarily gendered language (e.g. uses "they" singular rather than "he or she" for an unknown person).
* [ ] Informative link text (e.g. instead of "To learn more about python, click [here](www.example.com)", say "Read this article to [learn more about python](www.example.com).")
* [ ] Includes accurately formatted and functional link to feedback form.
* [ ] Spelling and grammar are correct.

## Organization

* [ ] Clear, informative headers and sensible hierarchical structure (the TOC in the left margin should give a good overview of the content convered)
* [ ] Uses specially formatted highlight boxes consistently and appropriately
* [ ] Short, digestible pieces --- avoids long paragraphs and breaks long sections up with sub-headers

## Formative assessment

* [ ] Frequent [formative assessment](https://carpentries.github.io/instructor-training/02-practice-learning/#identifying-and-correcting-misconceptions) in the form of knowledge checks and/or hands-on exercises
* [ ] Clear explanations available after questions unless the nature of the question itself or answer options makes it unnecessary (e.g. a T/F question may not always require follow-up explanation)
* [ ] Knowledge check questions and hands-on exercises relate directly to learning objectives

## Videos and images

* [ ] Screencasts cover a single coherent task so the recording is a short as is feasible. To demonstrate more than one related task, include several short screencasts in succession rather than recording one long screencast.
* [ ] Subtitles available for every recording with audio.
* [ ] Alt text available for every image.
* [ ] Important visuals (in video, image, or gif) are always described in the audio or in accompanying text.
  - For example, in a screencast, instead of just, "And then click here," provide description that could help scaffold someone without visual access like, "And then click on the button that says 'Run' in the top-right corner of the screen". Be sure to make use of text cues when available (e.g. button labels), not just visual signals like color or location.
  - When important content is conveyed in a visual, describe the key elements. For example, "Running this query produces the table below. It displays the first 5 rows by default, and columns for ID, encounter ID, diagnosis, and outcome."
  - When including a data visualization, describe important features, such as both axis labels and visible trends in the data. For example, "Here's a scatterplot showing number of encounters on the y-axis and age on the x-axis. All 183 patients from our sample are represented here, and it looks like a weak positive trend, with older patients being more likely to have had more encounters. There are a few important outliers, though, such as this patient at about 6 months old with more than 20 encounters already."
  - When visual information is repeated with minimal changes, it's fine to indicate that without providing a full description again. For example, "And here's the updated table, filtered to only show patients who have been seen in the last 2 years."
  - When important visual information in a video is too complex to include sufficient audio description (i.e. it would slow the content down so much as to impair its utility), an alternative video file should be provided with audio descriptions included.
* [ ] Color is never the sole method for distinguishing visual content (including in data visualizations).

## Branch References to Change prior to PR

List here any internal references (stated or hyperlinked) that work now because they refer to the named branch, but will not work once this is on the main branch and the named branch is deleted.

* [ ] description or quote, line ___ in file ____
* [ ] description or quote, line ___ in file ____
* [ ] description or quote, line ___ in file ____

**Comment by rosemm** [comment]
Hi @nafeldman! This looks awesome! I caught a couple little bugs, which I thought might be faster and easier to fix with commits rather than making you go back in and edit --- I pushed commits for those, please take a look and of course feel free to undo any changes you don't like. It's all super minor stuff, though, so hopefully all good. 
The one other change I was wondering about is your images. I really like them, and I really like the alt text for them; I feel like it actually adds kind of a lot to the module to have access to some of those descriptions, which makes me wonder if maybe we want to move them to image captions instead of alt text. Implementing that is a little tricky, though, because 1) we shouldn't repeat content as alt text and caption for the same image (that makes it read twice in a screen reader, which is annoying), but 2) we do still want genuine alt text for each image, even if we're also including a caption. So I'm imagining that maybe we could revise for each image on a case-by-case basis, trying to split genuine alt text from commentary that could be usefully moved to a caption. For example, "Line illustration of the increasingly hard to find CD/DVD Drive" could be revised to alt text "Line illustration of CD/DVD drive" and caption "The increasingly hard to find CD/DVD Drive" or even "Good luck finding one of these". Some of the current alt text reads to me like just pure alt text (e.g. "Folder icon with multiple nested documents"), with no content that someone might benefit from in a caption, so those we could leave caption-less (or add new captions, if you have an additional comment you'd like to add for them!). Let me know what you think!

**Comment by nafeldman** [comment]
thank you @rosemm I will go back in and add captions where it makes sense. I also noticed my additional resources that I thought I added are missing (evincing perils on working on your desktop) so I will add that back in too.

**Comment by rosemm** [comment]
Hi @nafeldman! It looks like you did accidentally break some of the images --- quick fix, though! :) You added backticks around several of them, which prevents them from rendering as images and just shows the literal markdown code instead. So we'll need to remove those backticks again, and then edit to add captions where you want them. To add a caption, you would edit something like this: 
`![Anarchic looking Windows desktop with far too many many icons and applications.](media/messy_desktop.png)`
(currently this is an image with alt text "Anarchic looking Windows desktop with far too many many icons and applications.") to something like this:
`![Windows desktop with many icons and applications.](media/messy_desktop.png "If this anarchic looking desktop feels familiar, you're likely relying on local storage.")`
(This second example now has alt text "Windows desktop with many icons and applications." and a caption that reads "If this anarchic looking desktop feels familiar, you're likely relying on local storage.") 
I know this is fiddly markdown --- if it would be a help, I'd be happy to go through and do this stuff so you don't have to spend time on it! You've already done such a beautiful job putting the module together, I hate to have you fiddle around with the little formatty things now. 
One other small thing: I saw you edited the answer divs immediately after quiz questions back to starting with some blank lines and then `---` instead of `***`. For them to show up correctly (i.e. only after the learner answers the question), there can't be any blank lines and it has to start with the same `***` that the answer div ends with. I'm happy to go in and fix that if you like.

**Comment by rosemm** [comment]
Oh, and thanks for adding the additional resources! They look great.

**Comment by nafeldman** [comment]
@rosemm if you don't mind making the formatting changes that would be great, I really appreciate it. I do my editing in VSCode and I think that it revert and break some formatting things from my experiences working with @leemc-data-ed as well.

**Comment by rosemm** [comment]
Sure thing! Happy to. :) I should be able to do that this afternoon. 
Please do feel free to make any changes you like to the captions I come up with --- I'll try to stick to wording from your alt texts when possible, but I think I'll have to do some editing for clarity etc. and I don't want to put words in your mouth, so don't feel at all shy about changing what I suggest.

**Comment by rosemm** [comment]
Okie dokie, I've got it updated now! Take a look and let me know if it looks okay to you. 
Note that we just changed our system for naming quizzes (sorry for swapping horses midstream on you!), so I also updated the names of quiz sections to reflect that new policy. If you're happy with it at this point, I think it's ready to merge!

**Comment by nafeldman** [comment]
@rosemm your changes look awesome, thank you! feel free to merge.
